crypto/fipsmodule/digest/internal.h

```diff
@@ -79,7 +79,11 @@ struct env_md_st {
   void (*init)(EVP_MD_CTX *ctx);
 
   // update hashes |len| bytes of |data| into the state in |ctx->md_data|.
-  void (*update)(EVP_MD_CTX *ctx, const void *data, size_t count);
+  // Digest update functions always return 1. update calls after |final| are 
+  // restricted via |ctx| check (|final| cleanses the |ctx|). Digest XOF update
+  // function propagates the return value from |SHAKE_Absorb|, that is 1 on
+  // success and 0 on failure, to restrict update calls after |squeezeXOF|.
+  int (*update)(EVP_MD_CTX *ctx, const void *data, size_t count);
```

Comment:
NP: I think mentioning `SHAKE_Absorb` in the description bothers me a little. Perhaps it can be slightly more generic:
```
// Returns 1 on success and 0 on failure. Failures can only occur on a 
// digest XOF update if called after |squeezeXOF|.
```